features/FEATURE_LWIP/lwip-interface/lwip-sys/arch/lwip_sys_arch.c

```diff
@@ -123,15 +123,15 @@ u32_t sys_now(void) {
  *---------------------------------------------------------------------------*/
 err_t sys_mbox_new(sys_mbox_t *mbox, int queue_sz) {
     if (queue_sz > MB_SIZE)
-        error("sys_mbox_new size error\n");
+        MBED_ERROR(MBED_MAKE_ERROR(MBED_MODULE_NETWORK_STACK, MBED_ERROR_CODE_INVALID_SIZE), "sys_mbox_new size error\n", queue_sz);
```

Comment:
I would recommend not to replace API in library/feature code (even when deprecating), let the feature owners do it after thorough testing.

Reply:
You can share the samples and examples, but actual API usage should be done by feature owners.

Reply:
@deepikabhavnani - Lot of these changes are already talked about during design review. In addition we are going to write separate requirements for each team once this is in. So, my understanding it is better to capture more info as long as they are not changing the behavior in any way.

Reply:
Yes Senthil, I believe it should be done and verified by feature owners